packages/hub/src/scope.ts

```diff
@@ -212,11 +212,7 @@ export class Scope implements ScopeInterface {
    * @inheritDoc
    */
   public getTransaction(): Transaction | undefined {
-    const span = this.getSpan() as Span & { spanRecorder: { spans: Span[] } };
-    if (span && span.spanRecorder && span.spanRecorder.spans[0]) {
-      return span.spanRecorder.spans[0] as Transaction;
-    }
-    return undefined;
+    return this.getSpan() as Transaction;
```

Comment:
This seems to be changing behavior in that if a span was stored in the scope, calling `getTransaction` would return the root span, a.k.a. the transaction.

Reply:
Same thing. The first thing we do when we initialize a span recorder on a transaction is to add the transaction itself as the first span. See https://github.com/getsentry/sentry-javascript/blob/8fcabd880eafc4f58581afffcc0d185c53122aa7/packages/tracing/src/idletransaction.ts#L187 and https://github.com/getsentry/sentry-javascript/blob/8fcabd880eafc4f58581afffcc0d185c53122aa7/packages/tracing/src/transaction.ts#L54.

So not only was that not necessary here, it also prevented us from finding `sampled.= false` transactions, since they don't have a span recorder.

Reply:
H:

I'll bring this back because I still think is is breaking existing behavior.

The `span.spanRecorder.spans[0] as Transaction` part being removed is responsible for fetching the transaction that contains the current span stored in the scope.

With this PR in, this method becomes "return whatever span is stored in the scope, pretend it is a transaction".

Reply:
This ties into the perennial question of whether transactions are truly a separate thing from spans or not. IMHO, we shouldn't be storing spans on the scope at all, but rather transactions. (I mean, we are, but we still call it `setSpan()` when we set it even though we call it `getTransaction()` when we get it, which is (maybe?) part of what's leading to this question.)

In the meantime, I'm still not clear how the two versions could ever lead to different results (other than the intended change, of non-sampled transactions no longer being ignored here). Because of the behavior of `initSpanRecorder` linked above, `mySpan.spanRecorder.spans[0]` _is_ `mySpan`. (And of course, that span is actually a transaction, see first paragraph.)

So I'm sorry, I'm not trying to be obtuse here, but I truly don't see where the potential problem is.

Reply:
After IRL convos, going to revert this change for now.